File: 0014-longest-common-prefix/0014-longest-common-prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        a = ""
        j = len(strs)
        l = 0
        b = ""
        for _ in range(len(strs[0]  )):
            j = 0
            a = strs[0][0:_+1]
            for st in strs:
                if a == st[0:_+1]:
                    j+=1
            if j == len(strs):
                    b = a
            else:
                    return b

        return b

File: 0014-longest-common-prefix/0014-longest-common-prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        string ss = strs[0];
        for (auto x : strs) {
            while (ss.size() > 0) {
                int v = ss.size();
                x.resize(v);
                if (x == ss) break;
                else {
                    int j = v-1;
                    ss.resize(j);
                }
            }
            if (ss.size() == 0)
                return "";
        }
        return ss;
    }
};